cognite/neat/_rules/models/dms/_validation.py

```diff
@@ -334,6 +339,51 @@ def _validate_value_type_existence(self) -> None:
                     )
                 )
 
+    def _validate_reverse_connections(self) -> None:
+        # do not check for reverse connections in Cognite models
+        if self.metadata.as_data_model_id() in COGNITE_MODELS:
+            return None
+
+        properties_by_view = {f"{prop_.view.suffix}.{prop_.property_}": prop_ for prop_ in self.properties}
+        reversed_ids = {
+            id_
```

Comment:
Any reason not to store the reverse property here? since you are looking it up in the next step. 
```suggestion
            id_: prop_.connection

```
Then you don't have to do the casting in the loop below either as `reverse_ids: dict[str, ReverseConnectionEntity]`